commands/generate.go

```diff
@@ -0,0 +1,108 @@
+// Copyright (c) Alex Ellis 2017. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+package commands
+
+import (
+	"fmt"
+
+	"github.com/openfaas/faas-cli/schema"
+	"github.com/openfaas/faas-cli/stack"
+	"github.com/spf13/cobra"
+	yaml "gopkg.in/yaml.v2"
+)
+
+const (
+	functionNamespace = "openfaas-fn"
+	resourceKind      = "Function"
+	defaultAPIVersion = "openfaas.com/v1alpha2"
+)
+
+var api string
+
+func init() {
+	generateCmd.Flags().StringVar(&api, "api", defaultAPIVersion, "OpenFaaS CRD API version")
+	faasCmd.AddCommand(generateCmd)
+}
+
+var generateCmd = &cobra.Command{
+	Use:   "generate --api=openfaas.com/v1alpha2 --yaml stack.yml",
+	Short: "Generate Kubernetes CRD YAML file",
+	Long:  `The generate command creates kubernetes CRD YAML file for functions`,
+	Example: `faas-cli generate --api=openfaas.com/v1alpha2 --yaml stack.yml | kubectl apply  -f -
```

Comment:
is specifying openfaas.com required for the API at this point, do we expect that to change for the CRD?

Reply:
it was `o6s.io` and we've changed to `openfaas.com` so who knows lol

Reply:
There needs to be a default so that we can make this command easy to use.